crates/shared/src/price_estimation.rs

```diff
@@ -443,9 +443,11 @@ pub struct Query {
     /// buy_token.
     pub in_amount: NonZeroU256,
     pub kind: OrderKind,
-    /// If this is `Some` the quotes are expected to pass simulations using the
-    /// contained parameters.
     pub verification: Option<Verification>,
+    /// Signals whether responses from that were valid on previous blocks can be
+    /// used to answer the query.
+    #[serde(skip_serializing)]
+    pub block_dependent: bool,
```

Comment:
Personally, I'm not a fan of adding the concept of "block_dependency" to the price query. It feels like we are overloading the `PriceEstimating` interface to do way too many things ATM and that native price estimation should use a fundamentally different interface to price estimation.

For example, for 1Inch, we could have:

```rust
pub struct OneInchClient { /* ... */ }

impl OneInchClient {
    fn with_block_header(current_block_stream: CurrentBlockStream) -> Self { /* ... */ }
}
```

Then, we could implement separate price estimator and native price estimator implementations:

```rust
struct OneInchPriceEstimator(OneInchClient);

impl OneInchPriceEstimator {
    pub fn new(client: OneInchClient, current_block_stream: CurrentBlockStream) -> Self {
        Self(client.with_block_stream(current_block_stream))
    }
}

impl PriceEstimating for OneInchPriceEstimator { /* ... */ }

struct OneInchNativePriceEstimator(OneInchClient);

impl OneInchNativePriceEstimator {
    pub fn new(client: OneInchClient) -> Self {
        Self(client)
    }
}

impl OneInchNativePriceEstimator for OneInchPriceEstimator { /* ... */ }
```

Under the hood, they can share some methods on `OneInchClient` for doing price estimation, without having to add flags for controlling function behaviour.

This is really a nit, that requires a larger change, so I don't think it is needed for this PR, just wanted to share that I do believe that the changes here reinforce that we shouldn't be overloading the `PriceEstimating` interface for both quoting and native price estimates.